### DISCUSSION TITLE
Neural network for federated learning PoC

### DISCUSSION THREAD

**Discussion by hyunho-mo** [MyDigiTwinNL/MyDigiTwin-architecture-documentation#3] [General]
Hi @hcadavid and Sonja,

Today, we talked about the use of neural networks with federated learning on the splitter Lifelines data.
[EMC_MyDigiTwin_Consortium_Meeting_DL_2023.pptx](https://github.com/user-attachments/files/16361724/EMC_MyDigiTwin_Consortium_Meeting_DL_2023.pptx)

IMO, the most important thing to discuss and decide is "What do we want to predict" (which is relevant to how to label the data and what the objective function looks like). 
For instance, let's suppose our prediction target is "risk categories", then the task for NN is 'classification'; we can implement an MLP in an hour by using cross-entropy loss as our objective function.  

However, we have a problem narrowing our task down to this classification task since we do not have the ground truth for our (federated) supervised learning. The only ground truth (strictly speaking, it is not the ground truth, though) we have is self-reported event occurrence and estimated response date. 

With the given ground truth, we can make a NN for classification tasks that predict the occurrence of an event within a fixed time window (e.g., for each instance, set 1 if it has an event within 5 years; set 0 otherwise). In this case, it will be very easy to label the data and implement a NN model (only if we exclude instances with incomplete observations; i.e., only include individuals who responded to follow-up surveys), but in this case, our work would not have much clinical implications; although it can be used for showing the PoC of federated learning for MDT)

In my previous work, i evaluated a neural network for survival analysis (via time-to-event data representation) on Lifelines. The network itself (with a dedicated objective function) is proposed by this [paper ](https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/s12874-018-0482-1) and you can find its pytorch implementation [here ](https://github.com/czifan/DeepSurv.pytorch)(my python scripts are also based on it). In this work, we label the data with occurrence and time-to-event after right-censoring, and the NN model is used to estimate a parameter of the survival function, which enables the calculation of time-to-event. Considering that this kind of prediction task/model is complicated and requires much explanations, i am not sure it will be the best option we can choose. 

I attached the presentation note for the details of my previous work.
It would be great to discuss "What do we want to predict"

**Comment by KasiaSmietanka** [comment]
As mentioned during the meeting and discussed with others, I propose to predict the 10 years of the follow up of the below mentioned endpoints:

Proposed outcomes:
CVD ( a composite of MI, HF, and Stroke)
MI
HF
Stroke

**Reply by hyunho-mo** [reply]
I understood the outcome and endpoint definitions. What I would like to clarify is the prediction model's output in numerical value; Could you please explain?

**Comment by sonjageorgievska** [comment]
Thank you for your input. Currently, we are using scikit-learn for our implementation. However, if we decide to use PyTorch for the proof-of-concept, we would have more options. One such option is simulating a Fully Connected Network  with a Convolutional Neural Network that allows us to explicitly mask out the missing values per patient. This approach eliminates the need to impute missing data with artificial values. We will consider this option depending on our time constraints over the next two months.

**Reply by hyunho-mo** [reply]
That's an interesting idea for coping with missing variables! The choice of deep learning library may depend on the software/framework for federated learning; @hcadavid, do you have any option (regarding federated learning framework) already in mind?

**Reply by sonjageorgievska** [reply]
The federated learning is agnostic to the DL framework, as we are merely sending serialized data over the network. We only wanted to keep it simple for the moment. And yet to show the potential :D.

**Reply by Tevien** [reply]
CNNs also give us the opportunity of doing something novel when it comes to simulations.
